### ISSUE TITLE
pyscf module changes the logging time

### ISSUE THREAD

**Issue by whu-dft** [pyscf/pyscf#1039]
When I import pyscf (version=1.7.6)  as a module, the logging time will be changed:

Without importing pyscf, the logging time is correct:

```python
import logging

logger = logging.getLogger(__name__)
logger.setLevel(logging.INFO)
log_handler = logging.StreamHandler()
log_handler.setFormatter(
    logging.Formatter("%(asctime)s - %(filename)s:%(lineno)d - %(message)s")
)
logger.addHandler(log_handler)
logger.info("666")
```
Outputs:
```bash
2021-08-24 09:25:59,796 - test_logtime.py:17 - 666
```

After importing pyscf module:

```python
import logging
import pyscf

logger = logging.getLogger(__name__)
logger.setLevel(logging.INFO)
log_handler = logging.StreamHandler()
log_handler.setFormatter(
    logging.Formatter("%(asctime)s - %(filename)s:%(lineno)d - %(message)s")
)
logger.addHandler(log_handler)
logger.info("666")
```

Outputs:
```bash
1970-03-17 00:21:03,611 - test_logtime.py:16 - 666
```

I have checked that the system time is correct. So what have pyscf done ?  The loging time in the following pic makes me uncomfortable:

![无标题](https://user-images.githubusercontent.com/38948614/130541187-3073a5c8-b38a-4f89-a1e8-d11ff312d5bf.png)

**Comment by glowthrower** [comment]
Coincidentally, I saw very similar behaviour in an application of my own just yesterday, and had been meaning to go looking for the cause myself.

I don't think this is an issue with PySCF itself. The reason I say this is that I use multiple machines, all with PySCF 1.7.6 installed (so identical code being executed as far as PySCF goes), but this behaviour only happens on some of them. For example, on my laptop (which runs Ubuntu 18.04), your example produces "normal" output:
```
2021-08-24 08:10:37,808 - script.py:11 - 666
```
If I run the exact same script on a machine running CentOS 7, I get
```
1970-01-15 20:19:07,675 - script.py:11 - 666
```
The obvious difference between the two would be the versions of dependent libraries that are installed. Just out of curiosity, which versions of `numpy`, `scipy`, and  `h5py` do you have installed?

**Comment by glowthrower** [comment]
OK, well, at least I can tell you what's happening, although I don't know how yet. The issue is that at some point during the `import pyscf` call, the `time` module is being somehow interfered with. Specifically, `time.time` is replaced with `time.perf_counter`. Check this out:
```
In [1]: import time

In [2]: time.time
Out[2]: <function time.time>

In [3]: time.perf_counter
Out[3]: <function time.perf_counter>

In [4]: import pyscf

In [5]: time.time
Out[5]: <function time.perf_counter>

In [6]: time.perf_counter
Out[6]: <function time.perf_counter>
```
Since the `logging` module uses `time.time` to get the strings for the logs, that's causing the weird output. The output of `perf_counter()` is a much lower number than `time.time()`, so when you convert it into a datetime string, it thinks we're back in the 1970s. See [this wikipedia article](https://en.wikipedia.org/wiki/Unix_time) if you're interested to know exactly why.

I have no idea why this is happening (yet) -- and again, note that this only happens on some environments for me. 

The PySCF logging code does touch both `time.time` and `time.perf_counter`, but I can't find a place where it *sets* them. 

However, I tried the above with `import numpy`, `import scipy`, and `import h5py`, and it doesn't happen. So maybe my first assessment was wrong and this is actually due to PySCF itself, somehow.

**Comment by whu-dft** [comment]
Thanks for your kindly reply. I test it on both Ubuntu 18.04 and Ubuntu 20.04, all comes with the weired time output. As for the versions:

```python
In [1]: import scipy

In [2]: scipy.__version__
Out[2]: '1.6.1'

In [3]: import numpy as np

In [4]: np.__version__
Out[4]: '1.20.1'

In [5]: import h5py

In [6]: h5py.__version__
Out[6]: '3.2.1'

In [7]: import pyscf
/home/fdd/anaconda3/envs/jax_cuda112/lib/python3.9/site-packages/pyscf/lib/misc.py:46: H5pyDeprecationWarning: Using default_file_mode other than 'r' is deprecated. Pass the mode to h5py.File() instead.
  h5py.get_config().default_file_mode = 'a'

In [8]: pyscf.__version__
Out[8]: '1.7.6'
```

The test case you provide is interesting. I think `pyscf` must have done something related to it. Generally speaking, it will not be a big problem. I just hope someone who are familiar with `pyscf` could solve it.

**Comment by sunqm** [comment]
In pyscf-1.7, time.time was replaced by time.perf_counter for python >= 3.8. This problem is properly fixed in pyscf 2.0. Pip wheels for pyscf-2.0 were not built. To get pyscf-2.0, the only choice is to compile the source code from github at the moment.

**Comment by whu-dft** [comment]
Thanks for your explaination. Closed.
